core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -116,7 +116,7 @@ class TransactionStateManager(brokerId: Int,
 
     // enforce disabled unclean leader election, no compression types, and compact cleanup policy
     props.put(LogConfig.UncleanLeaderElectionEnableProp, "false")
-    props.put(LogConfig.CompressionTypeProp, NoCompressionCodec)
+    props.put(LogConfig.CompressionTypeProp, UncompressedCodec.name)
```

Comment:
Might be worth getting this in separately since I'm pretty sure creation of the transaction state topic fails totally without this.